### PR TITLE
Regression practice 2.0

### PR DESCRIPTION
**Pull request by Ganling6666**

### CODE REVIEW

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+data_2s_extras_removed = subset(data_2s, select = -c(item_num, stud_ans, exam_key, corr, bloom_rating, complexity_rating_mean, complexity_rating_median, item_code, stem))
+data_2s_unique = unique(data_2s_extras_removed, incomparables = FALSE)
+View(data_2s_unique)
+data_2s_exam_score <- data_2s_unique %>% 
+  select(exam1, exam2, finalexam, course_fullid, ver, sex_id, urm_id, eop_id, fgn_id, experiment1, ta_sect, satverbal, satmath, high_sch_gpa, mastered_topics_initial_kc)
+  
+data_2s_exam_score_true <- na.omit(data_2s_exam_score)
+```
+###I would like to create a model that could implement the random effects based on the TA sections
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd:51]: There should be a space between "###" and "I" in order for this to be interpreted by markdown as a header.

**Review by cfcraig — APPROVED** [review]
looks good, only one minor typo to be aware of for future Rmd files :-)

**Comment by Ganling6666** [comment]
I have changed the TA sections and they should now be unique!

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#Include random effects (clustering)
+mod2 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + (1|ta_sect), 
+           data=master_true)
+summary(mod2)
+AIC(mod2)
+```
+```{r}
+AIC(mod1, mod2)
+```
+####Again, the change of AIC is much bigger than 2
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd:118]: You have to leave a space after the #### tag for markdown to interpret this line as a header.

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+```
+####I would like to create a model that could implement the random effects based on the TA sections
+```{r}
+#First, include sex_id as a control variable
+mod1 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + sex_id + (1|ta_sect), 
+           data=master_true)
+summary(mod1)
+AIC(mod1)
+
+#Include eop_id as a control variable
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd:82]: `eop_id` and other demographics variables aren't really controls in our work... they are groups designated by categorical variables that we care about in our models. We are trying to compare groups of interest (women, URMs, FGNs, EOPs) to their traditionally overserved counterparts. This is in contrast to the control variables which we want to include to make sure we compare students at the same "elevation"--- across all demo groups--- in terms of academic backgrounds, but we don't really care about the estimates that come out of the regression for the control variables. We do care about the outcomes for the groups of interest. We can discuss this more tomorrow.

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+copy_to <- paste0(importable_data_dir, "master_2s_small_deid_scaled")
+# CFC: If the file already exists in the target directory, the file will not be 
+# overwritten and this command will return `FALSE`.
+file.copy(copy_from, copy_to)
+```
+
+```
+## [1] FALSE
+```
+###Import and read the data
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md:92]: Need space after ###

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+
+```r
+#data_2s <- readRDS(paste0(metadata_dir, "master_2s_small_deid_scaled.rds"))
+data_2s <- readRDS("G:/Shared drives/CEG Two-Stage Exams Analysis/Ganling/two-stage-2/original-data/master_2s_small_deid_scaled.rds")
+```
+##Data wrangling
+####Creating unique rows and remove the ones that are not necessary for this data model
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md:99]: Need space after ####

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md [hunk 6]
```diff
@@ -0,0 +1,461 @@
...
+```r
+data_2s_extras_removed = subset(data_2s, select = -c(item_num, stud_ans, exam_key, corr, bloom_rating, complexity_rating_mean, complexity_rating_median, item_code, stem))
+data_2s_unique = unique(data_2s_extras_removed, incomparables = FALSE)
+View(data_2s_unique)
+data_2s_exam_score <- data_2s_unique %>% 
+  select(exam1, exam2, finalexam, course_fullid, ver, sex_id, urm_id, eop_id, fgn_id, experiment1, ta_sect, satverbal, satmath, high_sch_gpa, mastered_topics_initial_kc)
+  
+data_2s_exam_score_true <- na.omit(data_2s_exam_score)
+```
+##Make 'ta_sect'unique across different data values
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md:110]: Need space after ##

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md [hunk 7]
```diff
@@ -0,0 +1,461 @@
...
+####First, we need to split the data from into year 16 and year 17
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md:111]: Etc

File: processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md [hunk 8]
```diff
@@ -0,0 +1,461 @@
...
+####First, we need to split the data from into year 16 and year 17 
+
+```r
+data_2s_exam_score_true$ta_sect <- as.character(data_2s_exam_score_true$ta_sect)
+
+#next, split the dataframe in half
+master_unique_2016 <- subset(data_2s_exam_score_true, course_fullid != "CHEM_142_A_2017_4" & course_fullid != "CHEM_142_B_2017_4")
+master_unique_2017 <- subset(data_2s_exam_score_true, course_fullid != "CHEM_142_A_2016_4" & course_fullid != "CHEM_142_B_2016_4")
+```
+####Use the vectorization function in R, we can add '_16' and '_17'at the end of each TA section
```
Comment (cfcraig) [inline processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.md:120]: Etc

**Review by cfcraig — CHANGES_REQUESTED** [review]
Please add spaces after # header designations so markdown will interpret them correctly, then recompile the MD files. 

Your addition of the vectorized code looks great. 👌

**Pull request closed without merging**